chirp/drivers/tk690.py

```diff
@@ -0,0 +1,1211 @@
+
+        # adding it
+        self._banks[bank].append(loc - 1)
+
+        # if the update was successful, update in the memmap
+        self._update_bank_memmap()
+
+    def _del_channel_from_bank(self, loc, bank=None):
+        """Remove a channel from a bank, if no bank is specified we find
+        where it is"""
+
+        # if we don't know what bank channel is in,
+        if bank is None:
+            bank = self._get_bank(loc)
+
+        # in case memory entry isn't saved to any bank, no need to delete:
+        if bank is not None:
+            # remove it
+            self._banks[bank].pop(self._banks[bank].index(loc - 1))
+
+        # if the delete was successful update in the memmap
+        self._update_bank_memmap()
+
+    def _update_bank_memmap(self):
+        """This function is called whatever a change is made to a channel
+        or a bank, to update the memmap with the changes that was made"""
+
+        bl = b""
+        bb = b""
+
+        # group_belong index
+        gbi = 0
+        # sort the banks for consistient memory entries
+        self._banks = dict(sorted(self._banks.items()))
+        for bank in self._banks:
+            # check for empty banks
+            if len(self._banks[bank]) == 0:
+                bl += b"\xff\xff"
+                continue
+
+            # channel index inside the bank, starting at 1
+            # aka channel group index
+            cgi = 1
+            for channel in range(0, len(self._banks[bank])):
+                # update bb
+                bb += bytes([cgi, self._banks[bank][channel]])
+                # set the group limits for this group
+                if cgi == 1:
+                    bl += bytes([gbi, len(self._banks[bank])])
+
+                # increment both indexes
+                gbi += 1
+                cgi += 1
+
+        # fill the gaps before write it
+        bb += b"\xff" * 2 * int(self._num_banks - len(bb) / 2)
+        bl += b"\xff" * 2 * int(self._num_banks - len(bl) / 2)
+
+        # update the memmap
+        self._fill(0x1480, bl)
+        self._fill(0x1600, bb)
+
+    def get_settings(self):
+        """Translate the MEM_FORMAT structs into the UI"""
+        button_assignments = RadioSettingGroup("Button Functions",
+                                               "Configurable Button Functions")
+        group = RadioSettings(button_assignments)
+        for buttonName in ASSIGNABLE_BUTTONS:
+            _fullHeadWarning = ""
+            if buttonName in FULL_HEAD_ONLY_BUTTONS:
+                _fullHeadWarning = "  (If Equipped)"
+            rs = RadioSetting(
+              buttonName,
+              "Configured function for "+buttonName+" button"+_fullHeadWarning,
+              RadioSettingValueMap(
+                    BUTTON_FUNCTION_LIST,
+                    self._memobj.button_assignments[buttonName]))
+            button_assignments.append(rs)
+        return group
+
+    def set_settings(self, settings):
```

Comment:
This also looks like copy pasta from another driver, handling multiple overly-complex setting possibilities that aren't actually in the list returned from get_settings. I'm guessing L1161 is the only part of this that is actually used.